src/export/sealed.rs

```diff
@@ -0,0 +1,82 @@
+#[allow(unused_imports)]
+use crate as defmt;
+use crate::{Format, Formatter, Str};
+
+pub trait Truncate<U> {
+    fn truncate(self) -> U;
+}
+
+macro_rules! impl_truncate {
+    ($t:ty, $u:ty) => {
+        impl Truncate<$t> for $u {
+            fn truncate(self) -> $t {
+                self as $t
+            }
+        }
+    };
+}
+
+impl_truncate!(u8, u8);
+impl_truncate!(u8, u16);
+impl_truncate!(u8, u32);
+impl_truncate!(u8, u64);
+impl_truncate!(u8, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u16, u16);
+impl_truncate!(u16, u32);
+impl_truncate!(u16, u64);
+impl_truncate!(u16, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u32, u32);
+impl_truncate!(u32, u64);
+impl_truncate!(u32, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u64, u64);
+impl_truncate!(u64, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u128, u128);
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct NoneError;
+
+impl Format for NoneError {
+    fn format(&self, _fmt: Formatter) {
+        unreachable!();
+    }
+
+    fn _format_tag() -> Str {
+        defmt_macros::internp!("Unwrap of a None option value")
+    }
+
+    fn _format_data(&self) {}
+}
+
+pub trait IntoResult {
```

Comment:
It would be good to document/link what uses this API (since `export` is meant for use by macros, rust-analyzer's "Find References"  doesn't work).

For this trait, that would be the `defmt::unwrap!` macro.